Rules/ReviewUnusedParameter.cs

```diff
@@ -40,24 +40,22 @@ public IEnumerable<DiagnosticRecord> AnalyzeScript(Ast ast, string fileName)
                 IEnumerable<Ast> parameterAsts = scriptBlockAst.FindAll(oneAst => oneAst is ParameterAst, false);
 
                 // list all variables
-                IEnumerable<string> variables = scriptBlockAst.FindAll(oneAst => oneAst is VariableExpressionAst, false)
-                    .Cast<VariableExpressionAst>()
-                    .Select(variableExpressionAst => variableExpressionAst.VariablePath.UserPath);
+                IDictionary<string, int> variableCount = scriptBlockAst.FindAll(oneAst => oneAst is VariableExpressionAst, false)
+                    .Select(variableExpressionAst => ((VariableExpressionAst)variableExpressionAst).VariablePath.UserPath)
+                    .GroupBy(variableName => variableName)
+                    .ToDictionary(variableName => variableName.Key, variableName => variableName.Count());

```

Comment:
Dictionary should be case-insensitive

Reply:
Fixed in https://github.com/PowerShell/PSScriptAnalyzer/pull/1382/commits/9f61f2eab16818df510b176ffbff98ef78b91df5